### DISCUSSION TITLE
Question: It's not possible to pass Token key via query-param on wss URL

### DISCUSSION THREAD

**Discussion by bundasmanu** [deepgram/community#831] [General help] [answered]
Hi guys,

I'm new to deepgram.
And looking to the documentation, there is no way to pass API key via query-param in the wss url?
I'm question this, because i want to avoid the creation of a service to pass the Authorization via header.

Thanks

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
No, I'm sorry, but you cannot pass the API key as a query parameter in the WebSocket URL. Deepgram requires the API key to be passed in the Authorization header of the request. This is typically done in the format `Authorization: Token YOUR_DEEPGRAM_API_KEY`.

**Comment by nikolawhallon** [comment]
For Websockets running in browser/front-end apps though, one can specify the token in a `Sec-WebSocket-Protocol` let me try to dig up an example.

**Reply by lukeocodes** [reply]
Also, our SDK will sort out which way to do it automatically as it works on frontend and backend apps

**Reply by nikolawhallon** [reply]
`let socket = new WebSocket(‘wss://api.deepgram.com/v1/listen', [‘token’, ‘API_KEY’]);`
where this JS API is described here: https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications namely `webSocket = new WebSocket(url, protocols);`

**Comment by SandraRodgers** [comment]
@bundasmanu To summarize the answers:

- **Authorization Header (Standard Method)**
Deepgram requires the API key to be passed in the Authorization header.
Format: Authorization: Token YOUR_DEEPGRAM_API_KEY

- **Using Sec-WebSocket-Protocol Header (For Browser/Front-End Apps)**
You can specify the token directly in the WebSocket initialization.
Example:
```js
let socket = new WebSocket('wss://api.deepgram.com/v1/listen', ['token', 'YOUR_DEEPGRAM_API_KEY']);
```

- **Using the Deepgram SDK**
The SDK will handle the authorization process automatically.
Works for both front-end and back-end applications.

**Comment by bundasmanu** [comment]
Thank you so much,
Really good support here.

Understood. But does not exist a plan to allow in the future to set the token via query-param?

**Reply by lukeocodes** [reply, marked as answer]
Good question. Unlikely we'll support it in the query string.

Query strings are much more likely to be logged on servers without obfuscation, especially vanilla server configurations. 

Headers are less likely, especially ones like Authorization and Sec-WebSocket-*
